Lib/glyphsLib/builder/glyph.py

```diff
@@ -162,6 +163,11 @@ def to_ufo_glyph(self, ufo_glyph, layer, glyph, do_color_layers=True):  # noqa:
     self.to_ufo_hints(ufo_glyph, layer)  # .hints
     self.to_ufo_paths(ufo_glyph, layer)  # .paths
     self.to_ufo_components(ufo_glyph, layer)  # .components
+    # Store shape order for mixed glyphs
+    if layer.paths and layer.components:
+        ufo_glyph.lib[SHAPE_SIGNATURE_LIB_KEY] = "".join(
+            [("P" if isinstance(x, GSPath) else "C") for x in layer.shapes]
```

Comment:
maybe check that "C" is acutally a GSComponent? what if a new different of "shape" is introduced later that is neither GSPath nor GSComponent..